packages/ckeditor5-watchdog/tests/actionsrecorder.js

```diff
@@ -0,0 +1,424 @@
+
+			expect( insertContentRecord ).to.exist;
+			expect( insertContentRecord.params ).to.exist;
+		} );
+
+		it( 'should record nested operations with parent frames', () => {
+			editor.execute( 'paragraph' );
+
+			const records = plugin.getEntries();
+
+			// At least one record should have been created from the command execution
+			expect( records.length ).to.be.greaterThan( 0 );
+
+			// Check if any operations were recorded during command execution
+			const operationRecords = records.filter( record => record.action === 'model.applyOperation' );
+			if ( operationRecords.length > 0 ) {
+				expect( operationRecords.some( record => record.parentFrame ) ).to.be.true;
+			}
+		} );
+
+		it( 'should record view document events', () => {
+			const viewDocument = editor.editing.view.document;
+
+			viewDocument.fire( 'click', {
+				domEvent: new MouseEvent( 'click' ),
+				target: element
+			} );
+
+			const records = plugin.getEntries();
+			const clickRecord = records.find( record => record.action === 'observers:click' );
+
+			expect( clickRecord ).to.exist;
+		} );
+
+		it( 'should capture state snapshots correctly', () => {
+			editor.execute( 'paragraph' );
+
+			const records = plugin.getEntries();
+			const record = records[ 0 ];
+
+			expect( record.before ).to.have.property( 'documentVersion' );
+			expect( record.before ).to.have.property( 'editorReadOnly' );
+			expect( record.before ).to.have.property( 'editorFocused' );
+			expect( record.before ).to.have.property( 'modelSelection' );
+
+			expect( record.after ).to.have.property( 'documentVersion' );
+			expect( record.after ).to.have.property( 'editorReadOnly' );
+			expect( record.after ).to.have.property( 'editorFocused' );
+			expect( record.after ).to.have.property( 'modelSelection' );
+		} );
+
+		it( 'should record command execution results', () => {
+			class FooCommand extends Command {
+				isEnabled = true;
+
+				execute() {
+					return 'Foo result';
+				}
+			}
+
+			editor.commands.add( 'foo', new FooCommand( editor ) );
+			editor.execute( 'foo' );
+
+			const records = plugin.getEntries();
+			const commandRecord = records.find( record => record.action === 'commands.foo:execute' );
+
+			expect( commandRecord ).to.exist;
+			expect( commandRecord.result ).to.be.equal( 'Foo result' );
+		} );
+
+		it( 'should record errors during command execution', () => {
+			class FooCommand extends Command {
+				isEnabled = true;
+
+				execute() {
+					throw new Error( 'Test error' );
+				}
+			}
+
+			editor.commands.add( 'foo', new FooCommand( editor ) );
+
+			try {
+				editor.execute( 'foo' );
+			// eslint-disable-next-line no-unused-vars
+			} catch ( error ) {
+				// Expected error
+			}
+
+			const records = plugin.getEntries();
+			const commandRecord = records.find( record => record.action === 'commands.foo:execute' );
+
+			expect( commandRecord ).to.exist;
+			expect( commandRecord.error ).to.exist;
+		} );
+	} );
+
+	describe( 'record management', () => {
```

Comment:
Missing test(s) for `onMaxEntries` callback (include scenario for my comment in `src/actionsrecorder.ts`).